update lombok or make jdk9 compatible

For our  jdk9 builds we get an error:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by lombok.javac.apt.Processor (file:/home/travis/.m2/repository/org/projectlombok/lombok/1.12.6/lombok-1.12.6.jar) to field com.sun.tools.javac.processing.JavacProcessingEnvironment.processorClassLoader
```

Updating lombok improved the situation but still some powermock problems:

```
Caused by: java.lang.IllegalAccessError: class jdk.internal.reflect.ConstructorAccessorImpl loaded by org/powermock/core/classloader/MockClassLoader cannot access jdk/internal/reflect superclass jdk.internal.reflect.MagicAccessorImpl
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1007)
	at org.powermock.core.classloader.MockClassLoader.loadUnmockedClass(MockClassLoader.java:238)
```

And

```
Running de.komoot.photon.searcher.ElasticsearchSearcherFactoryTest
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.mockito.cglib.core.ReflectUtils$2 (file:/home/travis/.m2/repository/org/mockito/mockito-core/1.10.19/mockito-core-1.10.19.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of org.mockito.cglib.core.ReflectUtils$2
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.42 sec
```

Results:

```
Tests in error: 
  initializationError(de.komoot.photon.ReverseSearchRequestHandlerTest): java.lang.reflect.InvocationTargetException
  initializationError(de.komoot.photon.SearchRequestHandlerTest): java.lang.reflect.InvocationTargetException
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,3 @@ language: java
 jdk:
   - oraclejdk8
   - oraclejdk9
-  
-matrix:
-  allow_failures:
-  - jdk: oraclejdk9

--- a/pom.xml
+++ b/pom.xml
@@ -58,7 +58,7 @@
 		<dependency>
 			<groupId>com.beust</groupId>
 			<artifactId>jcommander</artifactId>
-			<version>1.32</version>
+			<version>1.48</version>
 		</dependency>
 		<dependency>
 			<groupId>org.openstreetmap.osmosis</groupId>
@@ -68,7 +68,7 @@
 		<dependency>
 			<groupId>org.apache.commons</groupId>
 			<artifactId>commons-lang3</artifactId>
-			<version>3.3.2</version>
+			<version>3.5</version>
 		</dependency>
 		<dependency>
 			<groupId>org.codehaus.jackson</groupId>
@@ -83,7 +83,7 @@
 		<dependency>
 			<groupId>org.projectlombok</groupId>
 			<artifactId>lombok</artifactId>
-			<version>1.12.6</version>
+			<version>1.16.18</version>
 			<scope>provided</scope>
 		</dependency>
 		<dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -58,7 +58,7 @@
 		<dependency>
 			<groupId>com.beust</groupId>
 			<artifactId>jcommander</artifactId>
-			<version>1.48</version>
+			<version>1.32</version>
 		</dependency>
 		<dependency>
 			<groupId>org.openstreetmap.osmosis</groupId>
@@ -68,7 +68,7 @@
 		<dependency>
 			<groupId>org.apache.commons</groupId>
 			<artifactId>commons-lang3</artifactId>
-			<version>3.5</version>
+			<version>3.3.2</version>
 		</dependency>
 		<dependency>
 			<groupId>org.codehaus.jackson</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -126,38 +126,8 @@
 		<dependency>
 			<groupId>org.mockito</groupId>
 			<artifactId>mockito-core</artifactId>
-			<version>1.10.19</version>
+			<version>2.13.0</version>
 			<scope>test</scope>
-		</dependency>
-
-		<dependency>
-			<groupId>org.powermock</groupId>
-			<artifactId>powermock-module-junit4</artifactId>
-			<version>1.6.4</version>
-			<scope>test</scope>
-			<exclusions>
-				<exclusion>
-					<groupId>org.hamcrest</groupId>
-					<artifactId>hamcrest-core</artifactId>
-				</exclusion>
-			</exclusions>
-		</dependency>
-
-		<dependency>
-			<groupId>org.powermock</groupId>
-			<artifactId>powermock-api-mockito</artifactId>
-			<version>1.6.4</version>
-			<scope>test</scope>
-			<exclusions>
-				<exclusion>
-					<groupId>org.mockito</groupId>
-					<artifactId>mockito-core</artifactId>
-				</exclusion>
-				<exclusion>
-					<groupId>org.hamcrest</groupId>
-					<artifactId>hamcrest-core</artifactId>
-				</exclusion>
-			</exclusions>
 		</dependency>
 
 		<dependency>

--- a/src/main/java/de/komoot/photon/searcher/AbstractPhotonRequestHandler.java
+++ b/src/main/java/de/komoot/photon/searcher/AbstractPhotonRequestHandler.java
@@ -20,7 +20,7 @@ public abstract class AbstractPhotonRequestHandler<R extends PhotonRequest> impl
     }        
 
     @Override
-    public final List<JSONObject> handle(R photonRequest) {
+    public List<JSONObject> handle(R photonRequest) {
         TagFilterQueryBuilder queryBuilder = buildQuery(photonRequest);
         Integer limit = photonRequest.getLimit();
         SearchResponse results = elasticsearchSearcher.search(queryBuilder.buildQuery(), limit);

--- a/src/main/java/de/komoot/photon/searcher/AbstractReverseRequestHandler.java
+++ b/src/main/java/de/komoot/photon/searcher/AbstractReverseRequestHandler.java
@@ -20,7 +20,7 @@ public abstract class AbstractReverseRequestHandler <R extends ReverseRequest> i
     }        
 
     @Override
-    public final List<JSONObject> handle(R photonRequest) {
+    public List<JSONObject> handle(R photonRequest) {
         TagFilterQueryBuilder queryBuilder = buildQuery(photonRequest);
         SearchResponse results = elasticsearchSearcher.search(queryBuilder.buildQuery(), photonRequest.getLimit(), photonRequest.getLocation(), 
                 photonRequest.getLocationDistanceSort());

--- a/src/test/java/de/komoot/photon/ReverseSearchRequestHandlerTest.java
+++ b/src/test/java/de/komoot/photon/ReverseSearchRequestHandlerTest.java
@@ -6,10 +6,8 @@
 package de.komoot.photon;
 
 import com.google.common.collect.ImmutableSet;
-import de.komoot.photon.query.BadRequestException;
-import de.komoot.photon.query.LanguageChecker;
-import de.komoot.photon.query.ReverseRequest;
-import de.komoot.photon.query.ReverseRequestFactory;
+import de.komoot.photon.query.*;
+import de.komoot.photon.searcher.ReverseRequestHandler;
 import de.komoot.photon.searcher.ReverseRequestHandlerFactory;
 import de.komoot.photon.searcher.SimpleReverseRequestHandler;
 import de.komoot.photon.utils.ConvertToGeoJson;
@@ -17,17 +15,13 @@ import org.hamcrest.core.IsEqual;
 import org.json.JSONObject;
 import org.junit.Assert;
 import org.junit.Test;
-import org.junit.runner.RunWith;
 import org.mockito.Mockito;
-import org.powermock.api.mockito.PowerMockito;
-import org.powermock.core.classloader.annotations.PowerMockIgnore;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
 import spark.Request;
 import spark.Response;
 
 import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Set;
 import org.elasticsearch.client.Client;
 import spark.RouteImpl;
@@ -37,9 +31,6 @@ import spark.RouteImpl;
  * @author svantulden
  */
 
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(SimpleReverseRequestHandler.class)
-@PowerMockIgnore({"javax.management.*"})
 public class ReverseSearchRequestHandlerTest {
     @Test
     public void testConstructor() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
@@ -61,11 +52,21 @@ public class ReverseSearchRequestHandlerTest {
         ReverseRequestFactory mockReverseRequestFactory = Mockito.mock(ReverseRequestFactory.class);
         Request mockWebRequest = Mockito.mock(Request.class);
         ReflectionTestUtil.setFieldValue(reverseSearchRequestHandlerUnderTest, ReverseSearchRequestHandler.class, "reverseRequestFactory", mockReverseRequestFactory);
-        ReverseRequestHandlerFactory mockReverseRequestHandlerFactory = Mockito.mock(ReverseRequestHandlerFactory.class);
+
+        SimpleReverseRequestHandler mockSimpleReverseRequestHandler = new SimpleReverseRequestHandler(null) {
+            @Override
+            public List<JSONObject> handle(ReverseRequest photonRequest) {
+                return new ArrayList<>();
+            }
+        };
+        ReverseRequestHandlerFactory mockReverseRequestHandlerFactory = new ReverseRequestHandlerFactory(null) {
+            @Override
+            public ReverseRequestHandler<ReverseRequest> createHandler(ReverseRequest request) {
+                return mockSimpleReverseRequestHandler;
+            }
+        };
         ReflectionTestUtil.setFieldValue(reverseSearchRequestHandlerUnderTest, ReverseSearchRequestHandler.class, "requestHandlerFactory", mockReverseRequestHandlerFactory);
-        SimpleReverseRequestHandler mockSimpleReverseRequestHandler = PowerMockito.mock(SimpleReverseRequestHandler.class);
-        PowerMockito.when(mockSimpleReverseRequestHandler.handle(Mockito.any(ReverseRequest.class))).thenReturn(new ArrayList<JSONObject>());
-        Mockito.when(mockReverseRequestHandlerFactory.createHandler(Mockito.any(ReverseRequest.class))).thenReturn(mockSimpleReverseRequestHandler);
+
         ConvertToGeoJson mockConvertToGeoJson = Mockito.mock(ConvertToGeoJson.class);
         ReflectionTestUtil.setFieldValue(reverseSearchRequestHandlerUnderTest, "geoJsonConverter", mockConvertToGeoJson);
         String expectedResultString = "{\"test\":\"success\"}";

--- a/src/test/java/de/komoot/photon/SearchRequestHandlerTest.java
+++ b/src/test/java/de/komoot/photon/SearchRequestHandlerTest.java
@@ -1,10 +1,12 @@
 package de.komoot.photon;
 
 import com.google.common.collect.ImmutableSet;
+import com.vividsolutions.jts.geom.Point;
 import de.komoot.photon.query.BadRequestException;
 import de.komoot.photon.query.LanguageChecker;
 import de.komoot.photon.query.PhotonRequest;
 import de.komoot.photon.query.PhotonRequestFactory;
+import de.komoot.photon.searcher.PhotonRequestHandler;
 import de.komoot.photon.searcher.PhotonRequestHandlerFactory;
 import de.komoot.photon.searcher.SimplePhotonRequestHandler;
 import de.komoot.photon.utils.ConvertToGeoJson;
@@ -12,25 +14,18 @@ import org.hamcrest.core.IsEqual;
 import org.json.JSONObject;
 import org.junit.Assert;
 import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.mockito.Mock;
 import org.mockito.Mockito;
-import org.powermock.api.mockito.PowerMockito;
-import org.powermock.core.classloader.annotations.PowerMockIgnore;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
 import spark.Request;
 import spark.Response;
-import spark.Route;
 
 import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Set;
 import org.elasticsearch.client.Client;
 import spark.RouteImpl;
 
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(SimplePhotonRequestHandler.class)
-@PowerMockIgnore({"javax.management.*"})
 public class SearchRequestHandlerTest {
     @Test
     public void testConstructor() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
@@ -52,11 +47,20 @@ public class SearchRequestHandlerTest {
         PhotonRequestFactory mockPhotonRequestFactory = Mockito.mock(PhotonRequestFactory.class);
         Request mockWebRequest = Mockito.mock(Request.class);
         ReflectionTestUtil.setFieldValue(searchRequestHandlerUnderTest, SearchRequestHandler.class, "photonRequestFactory", mockPhotonRequestFactory);
-        PhotonRequestHandlerFactory mockPhotonRequestHandlerFactory = Mockito.mock(PhotonRequestHandlerFactory.class);
+
+        SimplePhotonRequestHandler mockSimplePhotonRequestHandler = new SimplePhotonRequestHandler(null) {
+            public List<JSONObject> handle(PhotonRequest photonRequest) {
+                return new ArrayList<>();
+            }
+        };
+        PhotonRequestHandlerFactory mockPhotonRequestHandlerFactory = new PhotonRequestHandlerFactory(null) {
+            @Override
+            public PhotonRequestHandler<PhotonRequest> createHandler(PhotonRequest request) {
+                return mockSimplePhotonRequestHandler;
+            }
+        };
         ReflectionTestUtil.setFieldValue(searchRequestHandlerUnderTest, SearchRequestHandler.class, "requestHandlerFactory", mockPhotonRequestHandlerFactory);
-        SimplePhotonRequestHandler mockSimplePhotonRequestHandler = PowerMockito.mock(SimplePhotonRequestHandler.class);
-        PowerMockito.when(mockSimplePhotonRequestHandler.handle(Mockito.any(PhotonRequest.class))).thenReturn(new ArrayList<JSONObject>());
-        Mockito.when(mockPhotonRequestHandlerFactory.createHandler(Mockito.any(PhotonRequest.class))).thenReturn(mockSimplePhotonRequestHandler);
+
         ConvertToGeoJson mockConvertToGeoJson = Mockito.mock(ConvertToGeoJson.class);
         ReflectionTestUtil.setFieldValue(searchRequestHandlerUnderTest, "geoJsonConverter", mockConvertToGeoJson);
         String expectedResultString = "{\"test\":\"success\"}";

